### PR TITLE
Configuration-as-Code support

### PR DESCRIPTION
**Pull request by alecharp**
Only the view `ListViewColumn` is configurable from this plugin, other elements are used from other plugins.

### CODE REVIEW

File: src/test/java/com/cloudbees/plugins/deployer/casc/ConfigurationAsCodeTest.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    public JenkinsConfiguredWithCodeRule r = new JenkinsConfiguredWithCodeRule();
+
+    @Test
+    @ConfiguredWithCode("view-configuration.yml")
+    public void shouldBeAbleToConfigureAView() throws Exception {
+        final View view = r.jenkins.getView("view-with-deployNowColumn");
+        Assert.assertNotNull(view);
+
+        MatcherAssert.assertThat(view.getColumns(),
+              Matchers.hasItem(Matchers.instanceOf(DeployNowColumn.class))
```
Comment (varyvol) [inline src/test/java/com/cloudbees/plugins/deployer/casc/ConfigurationAsCodeTest.java:24]: Are the other columns there by default?

Comment (alecharp) [reply src/test/java/com/cloudbees/plugins/deployer/casc/ConfigurationAsCodeTest.java:24]: in the `all` view, this column is not by default, I created a view specifically to have this column. But it seems that my IDE validated the code, when it's in fact not compiling. Trying to clean that up.

File: pom.xml [hunk 2]
```diff
@@ -29,7 +29,7 @@
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>3.43</version>
+    <version>4.2</version>
```
Comment (MRamonLeon) [inline pom.xml:32]: 4.3 better?

Comment (alecharp) [reply pom.xml]: updated. Thanks.

**Comment by batmat** [comment]
Merge merge merge! :fireworks: 🙂

File: pom.xml [hunk 3]
```diff
@@ -134,15 +154,14 @@
...
 
     <!-- jenkins dependencies -->
 
     <dependency>
       <groupId>org.jenkins-ci.main</groupId>
       <artifactId>maven-plugin</artifactId>
-      <version>3.2</version>
+      <version>3.6</version>
       <exclusions>
         <exclusion>
```
Comment (oleg-nenashev) [inline pom.xml:166]: Is it still actual?

File: pom.xml [hunk 4]
```diff
@@ -157,21 +176,35 @@
...
+          <artifactId>hamcrest-core</artifactId>
+        </exclusion>
+      </exclusions>
     </dependency>
 
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>mailer</artifactId>
-      <version>1.17</version>
+      <scope>test</scope>
```
Comment (oleg-nenashev) [inline pom.xml:197]: Unrelated? it changes the dependency scope

Comment (alecharp) [reply pom.xml:197]: The diff shows I add the `test` scope, but it's not true. If you look at the end of the added line here, the scope was already set.

File: pom.xml (hunk 3 above)
Comment (alecharp) [reply pom.xml:166]: Haven't checked. Looking.

Comment (alecharp) [reply pom.xml:166]: not required. removing it.

File: pom.xml (hunk 4 above)
Comment (oleg-nenashev) [reply pom.xml:190]: Right, sorry

**Pull request merged**